proxy/src/http.rs

```diff
@@ -21,6 +24,23 @@ pub fn new_client() -> ClientWithMiddleware {
         .build()
 }
 
+pub fn new_client_with_timeout_and_retries(
+    default_timout: Duration,
+    total_retries: u32,
+) -> ClientWithMiddleware {
+    let timeout_client = reqwest::ClientBuilder::new()
+        .timeout(default_timout)
+        .build()
+        .expect("Failed to create http client with timeout");
+
+    let retry_policy = ExponentialBackoff::builder().build_with_max_retries(total_retries);
+
+    reqwest_middleware::ClientBuilder::new(timeout_client)
+        .with(reqwest_tracing::TracingMiddleware::default())
```

Comment:
I assume this is copy-pasta from `new_client` above?

Before this PR, the HTTP client we used didn't have it.

Let's keep to the subject and not add it in this PR.